test/integration/controller/trainjob_controller_test.go

```diff
@@ -123,6 +123,13 @@ var _ = ginkgo.Describe("TrainJob controller", ginkgo.Ordered, func() {
 			ginkgo.It("Should succeed to create TrainJob with TrainingRuntime", func() {
 				ginkgo.By("Creating TrainingRuntime and TrainJob")
 				gomega.Expect(k8sClient.Create(ctx, trainingRuntime)).Should(gomega.Succeed())
+				gomega.Eventually(func() error {
+					err := k8sClient.Get(ctx, client.ObjectKeyFromObject(trainingRuntime), trainingRuntime)
+					if err != nil {
+						return err
+					}
+					return nil
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
```

Comment:
```suggestion
				gomega.Eventually(func(g gomega.Gomega) {
					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(trainingRuntime), trainingRuntime)).Should(gomega.Succeeded())
				}, util.Timeout, util.Interval).Should(gomega.Succeed())
```

Let's use a propagated Gomega instance for better debaggability.